### PR TITLE
📖 Add deployment documentation for BMO and Ironic

### PR DESCRIPTION
**Pull request by furkatgofurov7**
**What this PR does / why we need it:**
This document describes the use cases of deploying baremetal-operator and Ironic.

### CODE REVIEW

File: docs/baremetal-operator-ironic-deployment.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
+# Running Bare Metal Operator with/without Ironic
```
Comment (dhellmann) [inline docs/baremetal-operator-ironic-deployment.md:1]: Maybe this file should just be called `deploying.md`?

**Comment by furkatgofurov7** [comment]
/continuous-integration/travis-ci/pr

File: docs/baremetal-operator-ironic-deployment.md (hunk 1 above)
Comment (furkatgofurov7) [reply docs/baremetal-operator-ironic-deployment.md]: Hi! Thanks for a review, updated the name of the document to `deploying`.

**Comment by furkatgofurov7** [comment]
/test-integration

> The instructions look OK. It would be useful to have some tips for when the different configurations are useful, too.

That would be definitely good to add, I have updated the document. Thanks!

File: docs/deploying.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+```ini
+
+$PROVISIONING_IP
+$PROVISIONING_CIDR
+$PROVISIONING_INTERFACE
+
+```
+
+In case you are deploying baremetak-operator locally, make sure to populate and
```
Comment (kashifest) [inline docs/deploying.md:122]: typo

Comment (furkatgofurov7) [reply docs/deploying.md:122]: Thanks, I will address all these feedbacks in #540

File: docs/deploying.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+   those to be deployed as a bundle.
+
+3. Only Ironic is deployed, in a case when BMO is deployed as part of CAPM3 and
+   only Ironic setup is sufficient, e.g.
+   [clusterctl](https://cluster-api.sigs.k8s.io/clusterctl/commands/move.html)
+   provided by Cluster API(CAPI) deploys BMO, so that it can take care of moving
+   the BaremetalHost during the pivoting.
+
+**Important Note**
+When the baremetal-operator is deployed through metal3-dev-env, this container
```
[the remark this answers is not in the document]
Comment (furkatgofurov7) [reply docs/deploying.md:111]: Thanks, I will address all these feedbacks in #540

File: docs/deploying.md [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+It is worth mentioning some tips for when the different configurations are
+useful as well. For example:
+
+1. Only BMO is deployed, in  a case when Ironic is already running, e.g. as part
+   of Cluster API Provider Metal3
+   [(CAPM3)](https://github.com/metal3-io/cluster-api-provider-metal3) when
+   a successful pivoting state was met and ironic being deployed.
+
+2. BMO and Ironic are deployed together, in a case when CAPM3 is not used and
+   those to be deployed as a bundle.
```
[the remark this answers is not in the document]
Comment (furkatgofurov7) [reply docs/deploying.md:102]: Thanks, I will address all these feedbacks in #540

**Pull request merged**